src/main/java/fishstock/task/Event.java

```diff
@@ -0,0 +1,99 @@
+package fishstock.task;
+
+import java.time.LocalDateTime;
+
+import fishstock.Command;
+import fishstock.UserInput;
+
+/**
+ * Encapsulates an Event Task.
+ * This Task has a description, from date, and to date.
+ */
+class Event extends Task {
+    private static final String FROM_KEYWORD = "/from";
+    private static final String TO_KEYWORD = "/to";
+    private LocalDateTime from;
+    private LocalDateTime to;
+
+    /**
+     * Initializes an Event object.
+     *
+     * @param description The task description.
+     * @param from The from date.
+     * @param to The to date.
+     */
+    protected Event(String description, boolean isDone, LocalDateTime from, LocalDateTime to) {
+        super(description, isDone);
+        this.from = from;
+        this.to = to;
+    }
+
+    protected Event(String description, LocalDateTime from, LocalDateTime to) {
+        this(description, false, from, to);
+    }
+
+    private static void checkIsValid(String[] splitInput) throws TaskException {
```

Comment:
Would it be better to use some sort of plural form for `splitInput`? I find it quite necessary to emphasise that the parameter is a collection of parts obtained by splitting the input.